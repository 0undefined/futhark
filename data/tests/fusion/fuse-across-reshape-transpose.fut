-- ==
-- input {
-- }
-- output {
--   [[2, 8, 14], [4, 10, 16], [6, 12, 18]]
-- }
-- structure { Map 2 }
fun main(): [][]int =
  let n = 9
  let a = map (+1) (iota(n))
  let b = reshape (3,3) a
  let c = transpose b in
  map  (fn (row: []int): []int  =>
         map  (*2) row) c
