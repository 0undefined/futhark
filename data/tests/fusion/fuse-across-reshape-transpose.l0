fun [[int]] main() =
  let n = 9 in
  let a = map(op+(1),iota(n)) in
  let b = reshape((3,3),a) in
  let c = transpose(b) in
  map (fn [int] ([int] row) =>
         map (op*(2), row),
       c)
