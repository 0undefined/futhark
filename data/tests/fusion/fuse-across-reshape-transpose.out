[[2, 8, 14], [4, 10, 16], [6, 12, 18]]
